src/rtt.c

```diff
@@ -354,28 +354,36 @@ void poll_rtt(target_s *const cur_target)
 	bool rtt_busy = false;
 
 	if (last_poll_ms + poll_ms <= now || now < last_poll_ms) {
-		target_addr_t watch;
-		target_halt_reason_e reason;
 		bool resume_target = false;
+
 		if (!rtt_found)
 			/* check if target needs to be halted during memory access */
 			rtt_halt = target_mem_access_needs_halt(cur_target);
+
+		target_addr_t watch;
 		if (rtt_halt && target_halt_poll(cur_target, &watch) == TARGET_HALT_RUNNING) {
 			/* briefly halt target during target memory access */
 			target_halt_request(cur_target);
-			while ((reason = target_halt_poll(cur_target, &watch)) == TARGET_HALT_RUNNING)
-				continue;
+
+			target_halt_reason_e reason;
+			do {
```

Comment:
would have preferred this to be a while loop w/ `reason` initialised to the truthy loop condition, but this works

Reply:
Ah, yeah you're right that didn't occur to me

Reply:
If you want to open a follow-up PR to change this, you'd be welcome to. Otherwise we'll address it when v1.10's cycle begins (assuming we remember)

Reply:
Yes i was thinking of that, I'll do one when i find a bit of dead time today